reVX/least_cost_xmission/README.md

```diff
@@ -1,25 +1,145 @@
 * [`least_cost_paths.py`](least_cost_paths.py) - Parent class for `least_cost_xmission.py`.
-* [`offshore_utilities.py`](offshore_utilities.py) - Utility functions and classes for preparing friction, barrier, and transmission features for offshore analyses. Example Jupyter notebooks for these functions can be found in the [`examples/least_cost_paths`](../../examples/least_cost_paths/) directory of this repository.
 
-<br>
+# Layer Creation
+Calculating transmission routing paths requires a series of layers. The *costs* layer is mandatory. The *friction* and *barriers* layers are optional. The *costs* layer is the most important of the three.
+
+* *Costs* - The cost of building a transmission line including ROW and any factors such as slope, land use, and water depth though a single grid cell (typically 90 meters.) The routing algorithm automatically scales costs for any transmission crossing a cell at a diagonal. Costs for land-based transmission and undersea cables are calculated separately and then combined. Note that landfall costs are incorporated into the final cost raster when wet and dry costs are combined.
+* *Friction* - Friction indicates areas that are undesirable for transmission. Paths that must cross a friction area will take the shortest possible path. Friction does not affect the cost of a line that runs across it; it merely discourages a path from crossing a friction area, possibly taking a longer and more expensive path.
+* *Barriers* - Barriers are effectively a much stronger deterrent for transmission than friction. Routing will preferentially take a more expensive, and higher friction route before crossing a barrier area. Barriers do not affect the cost of lines that are forced to cross them.
+
+Note that the *friction* and *barriers* layers must be combined together, using a multiplier for barriers, before an analysis is ran.
+
+Layers are created by passing a JSON configuration file (config file) to the [`transmission-layer-creator`](transmission_layer_creator_cli.py) command-line tool. The format of the JSON file is defined using [Pydantic](https://docs.pydantic.dev/latest/) in the `LayerCreationConfig` class of [`transmission_layer_creation.py`](../config/transmission_layer_creation.py). The config file consists of key-value pairs describing necessary files and layer creation options and can trigger a number of different operations depending on it's contents. These operations include:
+
+* Creating a new H5 file to store layers in
+* Creating wet, dry, and landfall masks from a vector
+* Creating wet and dry (TODO) cost layers
+* Combining wet and dry costs
+* Creating frictions and/or barrier layers
+* Combining friction and barrier layers together
+
+## Config File Keys
+### Required Keys
+The following keys are required to run any type of layer creation:
+
+* `template_raster_fpath` - Template GeoTIFF with the shape, CRS, and transform to use for the project. Data in the raster is ignored.
```

Comment:
Is there a better way to determine the profile for the layers we're creating other than using a template geotiff? Should there be one or two predefined profiles that the user can choose from?

Reply:
We should definitely allow users to provide their own profile (via a template raster or otherwise). Other than that, we *could* provide some defaults here, but that just sounds like extra work for you.

Maybe one thing in between would be also allowing the user to point to a cost/exclusion h5 that has a profile attribute that the code could pull?